server/aesdsocket.c

```diff
@@ -488,20 +648,60 @@ int main(int argc, char **argv)
         }
 
         // Logging in the client ip
+        //printf("\n");
         syslog(LOG_INFO, "Accepted connection from %s", client_ip);
-        //printf("[+] accepted client ip: %s\n", client_ip);
+       
+        ////printf("[+] accepted client ip: %s\n", client_ip);
+
+        // Creating a thread for accepted connection
+
+        thread_arg *t_args = malloc(sizeof(thread_arg));
+        if (!t_args) {
+            syslog(LOG_ERR, "Failed to allocate thread argument");
+            close(client_fd);
+            continue;
+        }
+
+        t_args -> client_fd = client_fd;
+        t_args -> file_fd = file_fd;
+        t_args -> socket_addr = client_addr;
 
-        int written_bytes;
-        // Receive packets from the client and store in SOCKETDATA_FILE
-        if ((written_bytes = send_rcv_socket_data(client_fd, file_fd)) > 0)
+        pthread_t thread_id;
+        syslog(LOG_INFO, "Creating a new thread");
+        int err = pthread_create(&thread_id, NULL, thread_operation, t_args);
+        if(err == 0)
         {
-            // Send back the stored data of file back to the client
-            return_socketdata_to_client(client_fd, file_fd);
+            //printf("[+] Thread creation Successfull \n");
         }
+        else{
+            //printf("[-] Thread creation Failed \n");
+            syslog(LOG_ERR, "Error creating thread: %s", strerror(err));
+            free(t_args);
+            close(client_fd);
+            continue;           // because we are accepting connection infinite times and seperating it to      multiple         threads 
+
+        }
+
+        thread_node_add(thread_id);
+        thread_join();
```

Comment:
@BhaktiRamani what does it mean to call thread_join here where `thread_join` unconditionally calls `pthread_join` for all threads regardless of whether they've completed?